### PR TITLE
support for non-elixir files with create_new_file, update_file, include_existing_file, include_or_create_file, create_or_update_file

### PR DESCRIPTION
**Pull request by mayel**
to start a discussion about how best to add support for patching non-elixir files

### CODE REVIEW

**Comment by zachdaniel** [comment]
This looks good to me 👍

**Comment by zachdaniel** [comment]
Happy to merge as is, but the title does say DRAFT, so LMK if its good to merge on your end :)

**Comment by mayel** [comment]
Great :) Marking as draft was because it needs a test, and maybe refactoring other functions in a similar way, such as`create_or_update_elixir_file` and `include_or_create_elixir_file` (adding `create_or_update_file` and `include_or_create_file`).

I see there's already `include_existing_file` and `update_file` functions but they includes a call to `format` so not sure if it's intended to work with non-elixir files?

**Comment by mayel** [comment]
Added a test, I guess the rest can also follow in other PRs

**Comment by zachdaniel** [comment]
Yeah, I think there is no need to load it up in one PR, happy to see separate PRs. Although, if you're going to do the other ones in the near future, we can leave it open and have it be one PR. Its your call 😄

**Comment by mayel** [comment]
Cool on it, easier to do while fresh than context switch :)

File: test/igniter/code/file_test.exs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+defmodule Igniter.Code.FileTest do
+  use ExUnit.Case
+
+  test "files will be created if they do not exist, in the conventional place, which can be configured" do
```
Comment (zachdaniel) [inline test/igniter/code/file_test.exs:4]: The conventional place for non-elixir files can't be configured, so I'd just remove the `igniter_exs` code and make the name of the test just "files will be created if they do not exist"

**Review by zachdaniel — COMMENTED** [review]
LGTM, one small requested change :)

**Comment by mayel** [comment]
done :)

**Pull request merged**